### PR TITLE
Move demo to react

### PR DESCRIPTION
**Pull request by timbset**

### CODE REVIEW

File: examples/todo/common/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "todo-common",
+  "version": "1.0.0",
+  "description": "",
+  "main": "./dist/index.js",
+  "author": "reimaginate team",
```
Comment (iokhotnikov) [inline examples/todo/common/package.json]: reimagined

File: examples/todo/server/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#
```
Comment (iokhotnikov) [inline examples/todo/server/.gitignore]: redundant line

File: examples/todo/client/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "todo-client",
+  "version": "0.1.0",
+  "private": true,
+  "proxy": "http://localhost:3001",
+  "dependencies": {
+    "bootstrap": "^3.3.7",
```
Comment (iokhotnikov) [inline examples/todo/client/package.json]: Looks like excessive dependency for simple example.

File: examples/todo/common/src/aggregates/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import Immutable from 'seamless-immutable';
+
+export default {
+    name: 'TodoItem',
+    initialState: () => null,
+
+    eventHandlers: {
+        TodoItemCreated: (state, event) => Immutable({ activated: true, cardId: event.cardId }),
+
```
Comment (iokhotnikov) [inline examples/todo/common/src/aggregates/TodoItem.js]: redundant line

File: examples/todo/common/src/aggregates/TodoCard.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import Immutable from 'seamless-immutable';
+
+export default {
+    name: 'TodoCard',
+    initialState: () => null,
+
+    eventHandlers: {
+        TodoCardCreated: () => Immutable({ activated: true }),
+
```
Comment (iokhotnikov) [inline examples/todo/common/src/aggregates/TodoCard.js]: redundant line

File: examples/todo/common/src/aggregates/TodoCard.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import Immutable from 'seamless-immutable';
+
+export default {
+    name: 'TodoCard',
+    initialState: () => null,
+
```
Comment (iokhotnikov) [inline examples/todo/common/src/aggregates/TodoCard.js]: redundant line

File: benchmarks/mongo-es-speed/buildState.js [hunk 7]
```diff
@@ -27,12 +29,18 @@ function reporterHandler() {
...
-    // eslint-disable-next-line no-console
-    console.log(DONE_TOKEN, JSON.stringify({
-        memory: process.memoryUsage()
-    }))
-).catch(err =>
-    // eslint-disable-next-line no-console
-    console.log(ERR_TOKEN, err)
-);
+store
+    .loadEventsByTypes(TYPES, () => eventCounter++)
```
Comment (IhostVlad) [inline benchmarks/mongo-es-speed/buildState.js]: Stairs-alike code. Function `loadEventsByTypes` and subsequent `then` and `catch` calls is not uniform. Function `then` is attached to `store.loadEventsByTypes` and has no relation to pure `store` object

File: examples/todo/client/src/components/TodoItem.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import classnames from 'classnames';
+
+export default class TodoItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { editing: false };
+        //     this._handleSave = (text) => {
```
Comment (iokhotnikov) [inline examples/todo/client/src/components/TodoItem.js]: Remove commented out code

File: benchmarks/mongo-es-speed/prepareEvents.js [hunk 9]
```diff
@@ -34,18 +36,17 @@ export default function (eventsCount, reportObj) {
...
-                payload: buildPayload(),
-                timestamp: Date.now()
-            }).then(() => ((++processedEvents % 5000 === 0)
-                ? (reportObj.value += 5000)
-                : null
-            ))
-        ),
+        acc =>
+            acc.then(() =>
+                store
```
Comment (IhostVlad) [inline benchmarks/mongo-es-speed/prepareEvents.js]: Stairs-alike code. `store` does not produce promise, which is chained in bottom code, but `store.saveEvents` does - so it should be monolith construction.

File: examples/todo/client/package.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    "react-media": "^1.5.1",
+    "react-redux": "^5.0.4",
+    "react-router-dom": "^4.1.1",
+    "redux": "^3.6.0",
+    "redux-burger-menu": "^0.2.0",
+    "redux-saga": "^0.15.3",
+    "resolve-redux": "*",
+    "seamless-immutable": "^7.1.2",
+    "socket.io-client": "^2.0.1",
+    "todo-common": "*"
```
Comment (iokhotnikov) [inline examples/todo/client/package.json]: Is it necessary use common code as package for simple example?

File: examples/todo/client/src/components/App.css [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+ * Base structure
+ */
+
+/* Move down content because we have a fixed navbar that is 50px tall */
+body {
+  padding-top: 50px;
+}
+
+
+/*
```
Comment (IhostVlad) [inline examples/todo/client/src/components/App.css]: Why variadic count of newlines before and after CSS classes and comments in this file?

File: examples/todo/client/src/components/TodoItem.js (hunk 8 above)
Comment (IhostVlad) [inline examples/todo/client/src/components/TodoItem.js]: Why commented code appeared? You can delete it and save in git history

File: benchmarks/mongo-es-speed/buildState.js (hunk 7 above)
Comment (timbset) [inline benchmarks/mongo-es-speed/buildState.js]: It was made by prettier. Its settings were discussed by the team

File: benchmarks/mongo-es-speed/prepareEvents.js (hunk 9 above)
Comment (timbset) [inline benchmarks/mongo-es-speed/prepareEvents.js]: And here prettier is

File: examples/todo/common/src/aggregates/TodoCard.js (hunk 5 above)
Comment (IhostVlad) [inline examples/todo/common/src/aggregates/TodoCard.js]: Why CRLF?

**Pull request merged**